pydantic/types.py

```diff
@@ -562,3 +564,49 @@ def validate(cls, value: NetworkType) -> Union[IPv4Network, IPv6Network]:
 
         with change_exception(errors.IPvAnyNetworkError, ValueError):
             return IPv6Network(value)
+
+
+class SecretStr(str):
```

Comment:
```suggestion
class SecretStr:
```

I don't think this should inherit from `str` since calling `str()` on it doesn't work as one might expect.